### DISCUSSION TITLE
Enquiry- Font, Terminal Copy/Paste in Safari, VI Compatibility

### DISCUSSION THREAD

**Discussion by Les-Linux** [coder/code-server#5814] [Q&A]
Hello,

First off, great job on code-server! I have a some questions I hope you can help me out with. I am running the powerlevel10 theme in ZShell and need to include some extra fonts. 

1. In MacOS, I simply drop them into ~/Library/Fonts, how do I make these available to my code-server terminal?
2. Copy / paste is not working for me in the terminal when accessed with Safari, however with Firefox it seems fine. Is there a KB article on how to get copy/paste in the terminal to work with Safari?
3. When I use vi to edit/view a file from the terminal I have to set :no compatible on each occasion or else the arrow key up/down will insert alpha characters rather than the intended mouse movement (up/down), is there a KB on how to have vi perform  as desired with code-server?

Keep up the good work.

Les

**Comment by lg-nineoneninethreenine** [comment]
I can answer Q1, but I don't know about the rest. I don't own a Mac; however the solution for your first question is general.

Go to Settings(UI) in code-server. Search for fonts and find the terminal font. Change that to MesloLGS Nerd Font, reload and cheerio.

**Reply by Les-Linux** [reply]
hey im-coder-lg.. that will only work if the font is already installed, unless I am mistaken?

**Reply by Les-Linux** [reply]
I also see 'The terminal only supports monospace fonts.' message thrown. Roboto Mono is a monospace font which also fails to be applied for me.

**Reply by lg-nineoneninethreenine** [reply]
That's weird. Let me check on my Linux instalation and report back.

By 7 PM IST mayhaps.

**Reply by jsjoeio** [reply]
The fonts should just work. I tested now on macOS and they do indeed update. Maybe that font isn't compatible? 

https://user-images.githubusercontent.com/3806031/204866521-be064f90-e894-4bd2-b6b1-164423977627.mov
